vcf-dumper/vcf-dumper-ws/src/test/java/uk/ac/ebi/eva/vcfdump/server/rest/VcfDumperIntegrationTest.java

```diff
@@ -0,0 +1,252 @@
+public class VcfDumperIntegrationTest {
+
+    private static final String EVA_ECABALLUS_20_DB = "eva_ecaballus_20";
+
+    private static final String EVA_NO_VARIANTS_DB = "eva_no_variants";
+
+    private static final String FILES_COLLECTION = "files";
+
+    private static final String VARIANTS_COLLECTION = "variants";
+
+    private static final String ANNOTATIONS_COLLECTION = "annotations";
+
+    private static final String ANNOTATIONS_METADATA_COLLECTION = "annotationsMetadata";
+
+    private static final int BLOCK_SIZE = 1000;
+
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Autowired
+    private MongoClient mongoClient;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Before
+    public void setUp() throws Exception {
+        createHorseDatabase();
+        createNoVariantsDatabase();
+    }
+
+    private void createHorseDatabase() throws IOException, URISyntaxException {
+        mongoClient.getDatabase(EVA_ECABALLUS_20_DB).drop();
+        MongoDatabase eva_ecaballus_20 = mongoClient.getDatabase(EVA_ECABALLUS_20_DB);
+        eva_ecaballus_20.createCollection(FILES_COLLECTION);
+        insertIntoCollectionFromFile(eva_ecaballus_20, FILES_COLLECTION, "test-data/files.json");
+        eva_ecaballus_20.createCollection(VARIANTS_COLLECTION);
+        insertIntoCollectionFromFile(eva_ecaballus_20, VARIANTS_COLLECTION, "test-data/variants.json");
+        eva_ecaballus_20.createCollection(ANNOTATIONS_COLLECTION);
+        insertIntoCollectionFromFile(eva_ecaballus_20, ANNOTATIONS_COLLECTION, "test-data/annotations.json");
+        eva_ecaballus_20.createCollection(ANNOTATIONS_METADATA_COLLECTION);
+        insertIntoCollectionFromFile(eva_ecaballus_20, ANNOTATIONS_METADATA_COLLECTION,
+                                     "test-data/annotationsMetadata.json");
+    }
+
+    private void insertIntoCollectionFromFile(MongoDatabase mongoDatabase, String collection, String path)
+            throws IOException, URISyntaxException {
+        ClassLoader classLoader = getClass().getClassLoader();
+        URI uri = Objects.requireNonNull(classLoader.getResource(path)).toURI();
+        String json = new String(Files.readAllBytes(Paths.get(uri)), StandardCharsets.UTF_8);
+        Document document = Document.parse(json);
+        mongoDatabase.getCollection(collection).insertOne(document);
+    }
+
+    private void createNoVariantsDatabase() {
+        mongoClient.getDatabase(EVA_NO_VARIANTS_DB).drop();
+        MongoDatabase eva_no_variants = mongoClient.getDatabase(EVA_NO_VARIANTS_DB);
+        eva_no_variants.createCollection(FILES_COLLECTION);
+        insertOneIntoFilesCollection(eva_no_variants, "1");
+    }
+
+    private void insertOneIntoFilesCollection(MongoDatabase mongoDatabase, String sid) {
+        Document file = new Document();
+        file.computeIfAbsent("sid", x -> sid);
```

Comment:
I was going to suggest `putIfAbsent` instead of `computeIfAbsent`, as I think the function `x -> sid` was unnecessary, but actually, you can simply `put`:
```suggestion
        file.put("sid", sid);
```